borg/archive.py

```diff
@@ -459,6 +460,15 @@ def rename(self, name):
         self.cache.chunk_decref(self.id, self.stats)
         del self.manifest.archives[self.name]
 
+    def set_comment(self, comment):
+        metadata = StableDict(self._load_meta(self.id))
+        metadata[b'comment'] = comment
+        data = msgpack.packb(metadata, unicode_errors='surrogateescape')
+        new_id = self.key.id_hash(data)
+        self.cache.add_chunk(new_id, data, self.stats)
+        self.manifest.archives[self.name] = {'id': new_id, 'time': metadata[b'time']}
+        self.cache.chunk_decref(self.id, self.stats)
+
```

Comment:
yeah, set_comment and rename should be deduplicated.

the common parts could move into a set_meta(key, value) function that is used by rename().
set_comment would be replaced by set_meta which would be called from archiver then.

please do an additional changeset for this and only collapse it after review.